index.js

```diff
@@ -161,36 +199,26 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
       Metadata: opts.metadata,
       StorageClass: opts.storageClass,
       ServerSideEncryption: opts.serverSideEncryption,
-      SSEKMSKeyId: opts.sseKmsKeyId,
-      Body: (opts.replacementStream || file.stream)
+      SSEKMSKeyId: opts.sseKmsKeyId
     }
 
     if (opts.contentDisposition) {
       params.ContentDisposition = opts.contentDisposition
     }
 
-    var upload = this.s3.upload(params)
-
-    upload.on('httpUploadProgress', function (ev) {
-      if (ev.total) currentSize = ev.total
-    })
-
-    upload.send(function (err, result) {
-      if (err) return cb(err)
+    if (file.mimetype !== 'image/svg+xml') {
+      params.Body = stream
+      var upload = this.s3.upload(params)
+      return initializeUploadEvents(upload, opts, cb)
+    }
 
-      cb(null, {
-        size: currentSize,
-        bucket: opts.bucket,
-        key: opts.key,
-        acl: opts.acl,
-        contentType: opts.contentType,
-        contentDisposition: opts.contentDisposition,
-        storageClass: opts.storageClass,
-        serverSideEncryption: opts.serverSideEncryption,
-        metadata: opts.metadata,
-        location: result.Location,
-        etag: result.ETag
-      })
+    file.buffer = ''
```

Comment:
i'll work on tests (sorry - stupid mistake :P) and npm modules. Did we want to do filesize checking in here? We never quite got to the planning of where to check filesize but i thought it might be somewhere outside this repo (before we read the whole file into the buffer)